leo/commands/build.rs

```diff
@@ -60,6 +62,18 @@ pub struct BuildOptions {
     pub enable_canonicalized_ast_snapshot: bool,
     #[structopt(long, help = "Writes AST snapshot after the type inference phase.")]
     pub enable_type_inferenced_ast_snapshot: bool,
+    #[structopt(long, help = "Writes all ASG snapshots for the different compiler phases.")]
+    pub enable_all_asg_snapshots: bool,
```

Comment:
Not sure about naming these snapshots. Snapshots refer to the AST's being written a specific way for ACL2 purposes. I think naming graph viz snapshots here could be confusing, especially when we add AST snapshots for constant folding & dead code elimination in the future. Probably worth a discussion, though.